src/main/java/de/craftlancer/clcapture/CapturePoint.java

```diff
@@ -222,22 +188,83 @@ private void runInactive() {
     }
     
     private void runActive() {
+        boolean areMultipleClans = false;
+        int scoreMultiplier = 1;
+        List<UUID> inRegionList = new ArrayList<>();
         lastTime = LocalTime.now().toSecondOfDay();
         
-        timeMap.replaceAll((a, b) -> a.equals(currentOwner) ? b + 1 : Math.max(b - 1, 0));
+        //If a player is within the capturepoint region, add them to the list.
+        Bukkit.getOnlinePlayers().forEach(a -> {
+            if (isInRegion(a))
+                inRegionList.add(convertToOwner(a));
+        });
         
-        bar.setProgress(timeMap.getOrDefault(currentOwner, 0) / (double) type.getCaptureTime());
-        bar.setTitle(name + " - " + getOwnerName());
+        //See if there are multiple clans by comparing clans
+        areMultipleClans = checkAreMultipleClans(inRegionList, areMultipleClans);
         
-        if (tickId % 20 == 0) {
-            bar.getPlayers().stream().filter(a -> a.getWorld().equals(signLocation.getWorld()))
-               .filter(a -> a.getLocation().distance(signLocation) >= type.getBossbarDistance()).forEach(bar::removePlayer);
-            Bukkit.getOnlinePlayers().stream().filter(a -> a.getWorld().equals(signLocation.getWorld()))
-                  .filter(a -> a.getLocation().distance(signLocation) < type.getBossbarDistance()).forEach(bar::addPlayer);
+        //If there are multiple clans present in the region, don't add to bossbar/map
+        if (!areMultipleClans) {
+            scoreMultiplier = handleSingleClanVariables(inRegionList, scoreMultiplier);

```

Comment:
This method does more than one thing it appears to do when looking at this code: getting the score multiplier.

Since the score multiplier would be obsolete with the change to Map from above you can probably change it to something like setOwner?